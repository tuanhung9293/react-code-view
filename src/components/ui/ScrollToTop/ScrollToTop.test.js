import React from 'react'
import {MemoryRouter} from 'react-router'
import {shallow, mount} from 'enzyme'
import ScrollToTop from './'

it('render ScrollToTop with a children', () => {
  expect(shallow(
    <ScrollToTop>I like Children</ScrollToTop>
  )).toMatchSnapshot()
})

it('onFetchCollectionUrl on componentDidUpdate if change props location.pathname', () => {
  const wrapper = mount(
    <MemoryRouter initialEntries={['/collection']}>
      <ScrollToTop>
          I like Children
      </ScrollToTop>
    </MemoryRouter>
  )
  
  wrapper.setProps({location: {
    pathname: 'pathname 2'
  }})

  expect(window.scrollTo).toBeDefined()
})
