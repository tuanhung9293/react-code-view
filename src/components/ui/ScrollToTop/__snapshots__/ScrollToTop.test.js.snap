// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ScrollToTop with a children 1`] = `
<Route>
  <Component />
</Route>
`;
